pkg/topology/kademlia/internal/metrics/metrics.go

```diff
@@ -164,7 +164,7 @@ type Counters struct {
 
 // flush writes the current state of in memory counters into the given db.
 func (cs *Counters) flush(db *shed.DB, batch *leveldb.Batch) error {
-	if cs.dirty.Load() > 1 {
+	if cs.dirty.Load() < 3 {
 		return nil
 	}
 	cs.dirty.CAS(3, 2)
```

Comment:
the CAS call here is now wrong since dirty will never be 3 anymore. it might be useful to sweep through the entire usage of this field to see that everything is correct

Reply:
How will it never be 3?  The earlier return is for those < 3, so only 3s will actually be coming through here.